// @milkshakechat/helpers/dist/types/base.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.placeholderAudioThumbnail = exports.placeholderImageThumbnail = exports.placeholderVideoThumbnail = exports.FirestoreCollection = void 0;
var FirestoreCollection;
(function (FirestoreCollection) {
    FirestoreCollection["USERS"] = "users";
    FirestoreCollection["PUSH_TOKENS"] = "pushTokens";
    FirestoreCollection["NOTIFICATIONS"] = "notifications";
    FirestoreCollection["FRIENDSHIPS"] = "friendships";
    FirestoreCollection["CHAT_ROOMS"] = "chatRooms";
    FirestoreCollection["STORIES"] = "stories";
})(FirestoreCollection || (exports.FirestoreCollection = FirestoreCollection = {}));
exports.placeholderVideoThumbnail = "https://firebasestorage.googleapis.com/v0/b/superlore-dev.appspot.com/o/defaults%2Fvideo-thumbnail.png?alt=media";
exports.placeholderImageThumbnail = "https://firebasestorage.googleapis.com/v0/b/superlore-dev.appspot.com/o/defaults%2Fimage-thumbnail.jpeg?alt=media";
exports.placeholderAudioThumbnail = "https://firebasestorage.googleapis.com/v0/b/superlore-dev.appspot.com/o/defaults%2Faudio-thumbnail.jpeg?alt=media";
