// @milkshakechat/helpers/dist/types/user.firestore.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultThemeColorHex = exports.privacyModeEnum = exports.genderEnum = exports.localeEnum = exports.FriendshipStatus = void 0;
var FriendshipStatus;
(function (FriendshipStatus) {
    FriendshipStatus["SENT_REQUEST"] = "SENT_REQUEST";
    FriendshipStatus["GOT_REQUEST"] = "GOT_REQUEST";
    FriendshipStatus["ACCEPTED"] = "ACCEPTED";
    FriendshipStatus["DECLINED"] = "DECLINED";
    FriendshipStatus["BLOCKED"] = "BLOCKED";
    FriendshipStatus["NONE"] = "NONE";
})(FriendshipStatus || (exports.FriendshipStatus = FriendshipStatus = {}));
var localeEnum;
(function (localeEnum) {
    localeEnum["english"] = "english";
    localeEnum["japanese"] = "japanese";
    localeEnum["spanish"] = "spanish";
    localeEnum["arabic"] = "arabic";
    localeEnum["korean"] = "korean";
    localeEnum["chinese"] = "chinese";
    localeEnum["thai"] = "thai";
    localeEnum["vietnamese"] = "vietnamese";
    // russian = "russian",
})(localeEnum || (exports.localeEnum = localeEnum = {}));
var genderEnum;
(function (genderEnum) {
    genderEnum["male"] = "male";
    genderEnum["female"] = "female";
    genderEnum["other"] = "other";
})(genderEnum || (exports.genderEnum = genderEnum = {}));
var privacyModeEnum;
(function (privacyModeEnum) {
    privacyModeEnum["public"] = "public";
    privacyModeEnum["private"] = "private";
    privacyModeEnum["hidden"] = "hidden";
})(privacyModeEnum || (exports.privacyModeEnum = privacyModeEnum = {}));
exports.defaultThemeColorHex = "#2EB8F6";
