// @milkshakechat/helpers/dist/types/wallet.firestore.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionStatus = exports.TransactionType = void 0;
var TransactionType;
(function (TransactionType) {
    TransactionType["COOKIE"] = "COOKIE";
    TransactionType["STICKER"] = "STICKER";
    TransactionType["GIFT_CARD"] = "GIFT_CARD";
    TransactionType["REFUND"] = "REFUND";
})(TransactionType || (exports.TransactionType = TransactionType = {}));
var TransactionStatus;
(function (TransactionStatus) {
    TransactionStatus["SUCCESS"] = "SUCCESS";
    TransactionStatus["FAILURE"] = "FAILURE";
})(TransactionStatus || (exports.TransactionStatus = TransactionStatus = {}));
/**
 * Example purchase of cookies
 * - 1 Transaction created
 * - 1 JournalEntries created
 *     - add cookies to receiver
 *
 * Example purchase of stickers
 * - 1 Transaction created
 * - 3 JournalEntries created
 *      - deduct cookies from sender
 *      - add cookies to receiver
 *      - add cookies to house
 *
 * Example purchase of gift card
 * - 1 Transaction created
 * - 1 JournalEntries created
 *      - add cookies to gift card
 *
 * Example redemption of gift card
 * - 1 Transaction created
 * - 2 JournalEntries created
 *      - deduct cookies from gift card
 *      - add cookies to receiver
 *
 */
