// @milkshakechat/helpers/dist/media/media-resized.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCompressedStoryImageUrl = exports.getCompressedWishlistGraphicUrl = exports.getCompressedStickerUrl = exports.getCompressedAvatarUrl = exports.BucketFolderSlug = exports.ImageResizeOption = void 0;
var ImageResizeOption;
(function (ImageResizeOption) {
    ImageResizeOption["thumbnail"] = "200x200";
    ImageResizeOption["compressed"] = "768x768";
})(ImageResizeOption || (exports.ImageResizeOption = ImageResizeOption = {}));
var BucketFolderSlug;
(function (BucketFolderSlug) {
    BucketFolderSlug["avatars"] = "avatars/";
    BucketFolderSlug["story_image"] = "story/image/";
})(BucketFolderSlug || (exports.BucketFolderSlug = BucketFolderSlug = {}));
function getCompressedAvatarUrl({ userID, assetID, size = ImageResizeOption.thumbnail, bucketName, }) {
    const base = `https://firebasestorage.googleapis.com/v0/b/${bucketName}/o/`;
    const path = encodeURIComponent(`users/${userID}/avatars/resized-media/${assetID}_${size}.jpeg`);
    const newURL = `${base}${path}?alt=media`;
    return newURL;
}
exports.getCompressedAvatarUrl = getCompressedAvatarUrl;
function getCompressedStickerUrl({ userID, assetID, size = ImageResizeOption.thumbnail, bucketName, }) {
    const base = `https://firebasestorage.googleapis.com/v0/b/${bucketName}/o/`;
    const path = encodeURIComponent(`users/${userID}/sticker/resized-media/${assetID}_${size}.jpeg`);
    const newURL = `${base}${path}?alt=media`;
    return newURL;
}
exports.getCompressedStickerUrl = getCompressedStickerUrl;
function getCompressedWishlistGraphicUrl({ userID, assetID, size = ImageResizeOption.compressed, bucketName, }) {
    const base = `https://firebasestorage.googleapis.com/v0/b/${bucketName}/o/`;
    const path = encodeURIComponent(`users/${userID}/wishlist/resized-media/${assetID}_${size}.jpeg`);
    const newURL = `${base}${path}?alt=media`;
    return newURL;
}
exports.getCompressedWishlistGraphicUrl = getCompressedWishlistGraphicUrl;
function getCompressedStoryImageUrl({ storyID, userID, size, assetID, bucketName, }) {
    const path = encodeURIComponent(`users/${userID}/story/IMAGE/${assetID}/resized-media/${assetID}_${size}.jpeg`);
    const base = `https://firebasestorage.googleapis.com/v0/b/${bucketName}/o/`;
    const newURL = `${base}${path}?alt=media`;
    console.log(`newURL`, newURL);
    return newURL;
}
exports.getCompressedStoryImageUrl = getCompressedStoryImageUrl;
// Actual
// https://firebasestorage.googleapis.com/v0/b/200x200/o/users%2Fm2fb0WWHOBesIAsevvCeNfv1w2Z2%2Favatars%2Fresized-media%2F92011d27-1d6e-453b-b994-d42a3d824657_200x200.jpeg?alt=media
// Expected
// https://firebasestorage.googleapis.com/v0/b/milkshake-dev-faf77.appspot.com/o/users%2Fm2fb0WWHOBesIAsevvCeNfv1w2Z2%2Favatars%2Fresized-media%2F92011d27-1d6e-453b-b994-d42a3d824657_200x200.jpeg?alt=media&token=18fb7d93-11e1-4999-88d2-abe5d8e91ee2
